File: Math/Spell Splice.cpp

// Problem link ===>>https://www.codechef.com/problems/CMIX
// submission link ===>> https://www.codechef.com/viewsolution/1063615305

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{
    ll t;
    cin >> t;
    while (t--)
    {
        ll n;
        cin >> n;

        vector<ll> v(n);
        vector<ll> a(n);
        for (ll i = 0; i < n;i++)
        {
            cin >> v[i] >> a[i];
        }
        ll ans = 0;
        for (ll i = 0; i < n; i++)
        {
            for (ll j = 0; j < n; j++)
            {
                if (i < j)
                {
                    ans = max(ans, (a[i] * v[j]) + (v[i] * a[j]));
                }
            }
        }
        cout << ans << endl;
    }
    return 0;
}

File: Math/Spell Splice.py

# Problem link ===>>


def main():
    import sys
    input = sys.stdin.read
    data = input().split()

    index = 0
    t = int(data[index])
    index += 1
    results = []

    for _ in range(t):
        n = int(data[index])
        index += 1

        v = []
        a = []

        for i in range(n):
            v.append(int(data[index]))
            a.append(int(data[index + 1]))
            index += 2

        ans = 0

        for i in range(n):
            for j in range(n):
                if i < j:
                    ans = max(ans, (a[i] * v[j]) + (v[i] * a[j]))

        results.append(ans)

    for result in results:
        print(result)


if __name__ == "__main__":
    main()
